src/theory/datatypes/inference_manager.h

```diff
@@ -25,6 +25,38 @@ namespace CVC4 {
 namespace theory {
 namespace datatypes {
 
+class DatatypesInference : public TheoryInference
```

Comment:
Why not inherit from `SimpleTheoryInternalFact`? It has exactly the same members (+ the proof generator). Don't we need the proof generator for `DatatypesInference` as well?

Reply:
Sure, this is a good idea (proof generators will be added later).